app/services/recipe.py

```
@@ -0,0 +1,113 @@
+            total_cost=analysis.total_cost,
+            servings=analysis.servings,
+            ingredients_data=ingredients_data,
+        )
+
+        return self._to_response(recipe)
+
+    async def save(self, recipe_id: str, user_id: str) -> RecipeResponse:
+        """레시피를 유저 컬렉션에 저장한다."""
+        recipe = await self.recipe_repo.find_by_id(recipe_id)
+        if not recipe:
+            raise NotFoundException("레시피를 찾을 수 없습니다")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)